feat: using DAP without nvim-dap-go

HeavyPunk opened the pull request:
Hello! I have a nvim config that contains a many of DAP configurations for some languages, also for golang. I don't want to install additional plugin special for debugging go apps or tests (this breaks a consistency of my config). I setup own DAP configuration for this and wanna use it to debug tests with your plugin.

fredrikaverpil commented:
Hey @HeavyPunk and many thanks for your contribution!

I think this is great and I actually have a colleague at work who wants to not to depend on sap-go as well.

I'm a little busy at the moment but let me take the time to review properly and then let's definitively allow manual DAP config. 👍

HeavyPunk commented:
Great! Also I give you example how this feature used in my config:
``` lua
require("neotest-golang") {
          dap_manual_enabled = true,
          dap_manual_configuration = {
            name = "Debug go tests",
            type = "go", -- preconfigured DAP adapter name
            request = "launch",
            mode = "test",
          }
        }
```

HeavyPunk commented:
Hey @fredrikaverpil ! I fixed a strategy selector to prevent loading dap-go and manual-dap modules before executing a logic. At now all of them are lazy-loaded.
This also fixed all tests

fredrikaverpil commented on lua/neotest-golang/features/dap/dap_manual.lua:
```diff
@@ -0,0 +1,45 @@
+local options = require("neotest-golang.options")
+local logger = require("neotest-golang.logging")
+
+local dap = require("dap")
```
Unused, looks like it can be removed.

fredrikaverpil commented on lua/neotest-golang/features/dap/dap_manual.lua:
```diff
@@ -0,0 +1,45 @@
+
+---Dummy function is needed to be corresponding to dap-go setup (just like trait implementation)
+function M.assert_dap_prerequisites()
+  logger.debug("Nothing to check. Manual DAP configuration in use")
```
Let's make this a no-op instead, no need to even log something here unless we can detect a problem.

fredrikaverpil reviewed (changes requested):
Great work @HeavyPunk!

I had some minor comments. I can implement these changes myself and get this merged if you like.

HeavyPunk commented:
Thank you for your review, @fredrikaverpil ! I implement your suggestions by myself. I call you here a bit later after fixing of code

HeavyPunk commented on README.md:
```diff
@@ -12,6 +12,7 @@ Reliable Neotest adapter for running Go tests in Neovim.
 - DAP support with [leoluz/nvim-dap-go](https://github.com/leoluz/nvim-dap-go)
   integration for debugging of tests using
   [delve](https://github.com/go-delve/delve).
+- DAP support for your own DAP configuration.
```
Hi!🖐 Yes, my example is enough for configuring DAP (only `type` field required additional info because it is user-specific parameter). I think this feature will be used by users who know how to configure DAP

HeavyPunk commented on tests/unit/options_spec.lua:
```diff
@@ -13,6 +13,8 @@ describe("Options are set up", function()
       go_list_args = {},
       gotestsum_args = { "--format=standard-verbose" },
       dap_go_opts = {},
+      dap_manual_enabled = false,
```
I think this is good idea. It will be more scaleable

HeavyPunk commented:
Hey @fredrikaverpil ! It's ready for your review 😀

fredrikaverpil commented:
Many thanks @HeavyPunk, I'll have a look and make any additional touch-ups to e.g. README but then I'll merge it in. Looks great, overall!

HeavyPunk commented:
Thanks! I will be waiting for merge :tada:

fredrikaverpil reviewed (approved):
🚀 super nice, thank you @HeavyPunk
